### PR TITLE
Completes 200 & 404 for GET /api/articles/:article_id

### PR DESCRIPTION
**Pull request by danrfletcher**

### CODE REVIEW

File: models/article-models.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const db = require('../db/connection.js');
+const format = require('pg-format');
+
+exports.selectArticle = (article_id) => {
+    return db.query(format(`SELECT * FROM articles WHERE article_id = %L;`, [article_id]))
+    .then((article) => {
+        console.log(article.rows[0])
```
Comment (MRombi) [inline models/article-models.js]: We should always remove console.logs 🙂

File: __tests__/app.test.js [hunk 2]
```diff
@@ -34,4 +34,27 @@ describe('GET Requests', () => {
...
+        })
+    });
+    test('/api/articles/:article_id 404: responds with a 404 error if article does not exist', () => {
+        return request(app)
+        .get('/api/articles/1000')
+        .expect(404)
+        .then((res) => {
+            expect(res.body.msg).toBe('article not found');
+        })
+    });
```
Comment (MRombi) [inline __tests__/app.test.js:59]: We should add a test for when the id is not a number 🙂

File: __tests__/app.test.js [hunk 3]
```diff
@@ -34,4 +34,27 @@ describe('GET Requests', () => {
             });
         });
     });
+    test('/api/articles/:article_id 200: responds with valid article object', () => {
+        return request(app)
+        .get('/api/articles/1')
+        .expect(200)
+        .then((res) => {
+            expect(typeof res.body.article_id).toBe('number');
```
Comment (MRombi) [inline __tests__/app.test.js]: We should check that the id is 1, being a number is not enough. We could also be more specific with the rest of the values 🙂

File: controllers/article-controllers.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const { selectArticle } = require('../models/article-models.js');
+
+exports.getArticleByID = (req, res, next) => {
+    const { article_id } = req.params;
+    selectArticle(article_id).then((article) => {
+        res.status(200).send(article)
```
Comment (MRombi) [inline controllers/article-controllers.js:6]: We should send the response back as a nested object with a key of article

**Review by MRombi — CHANGES_REQUESTED** [review]
Good job on your first pr for this endpoint! Please, action on the feedback I left and resubmit 🙂

**Comment by aloush-dev** [comment]
Awesome job implementing feedback.
Please remember to remove .only's before you resubmit a PR.

**Review by aloush-dev — APPROVED** [review]

**Pull request merged**